AndroidSDK/src/com/leanplum/internal/Request.java

```diff
@@ -383,11 +384,13 @@ private void sendNow() {
     Util.executeAsyncTask(true, new AsyncTask<Void, Void, Void>() {
       @Override
       protected Void doInBackground(Void... params) {
-        final List<Map<String, Object>> requestsToSend = getUnsentRequests();
+        List<Map<String, Object>> requestsToSend = getUnsentRequests();
         if (requestsToSend.isEmpty()) {
           return null;
         }
 
+        requestsToSend = updateUnsentRequests(requestsToSend);
```

Comment:
we should be sure that all requests were saved with generated UUID before sending them to the server.
We will have one UUID for the batch of actions, that means we cannot generate UUID when inserting. We can avoid double parsing with moving UUID logic to getUnsentRequests.